CMakeLists.txt

```diff
@@ -25,8 +25,12 @@ if(NOT HCC_BIN)
     message(FATAL_ERROR "Can't find hcc")
 endif()
 
-set(CLANG_BIN "${HCC_BIN}")
-set(BITCODE_DIR "${HCC_BIN}/../lib")
+if (NOT DEFINED CLANG_BIN)
+    set(CLANG_BIN "${HCC_BIN}")
```

Comment:
These should just be defined as a cache variable:

```cmake
set(CLANG_BIN "${HCC_BIN}" CACHE STRING "")
```

There is no need to check for if its defined. Since its a cached variable, it will not overwrite the users input.

Reply:
Okay that option works as well. I just tested in my example code:
```
  1 cmake_minimum_required(VERSION 3.5)
  2
  3 set(HCC_BIN "/opt/rocm/hcc/bin")
  4
  5 # if (NOT DEFINED CLANG_BIN)
  6   set(CLANG_BIN "${HCC_BIN}" CACHE STRING "")
  7 # endif()
  8 # if(NOT DEFINED BITCODE_DIR)
  9   set(BITCODE_DIR "${HCC_BIN}/../lib" CACHE STRING "")
 10 # endif()
 11
 12 message("--------CLANG_BIN: ${CLANG_BIN}")
 13 message("------BITCODE_DIR: ${BITCODE_DIR}")
 14
```